crates/wash-cli/src/cmd/wit/deps.rs

```
@@ -0,0 +1,321 @@
+                        path.display()
+                    );
+                }
+
+                match (ns, pkgs.as_slice()) {
+                    // namespace-level override
+                    (ns, []) => {
+                        wkg_config_overrides.insert(
+                            ns,
+                            Override {
+                                path: Some(path),
+                                version: None,
+                            },
+                        );
+                    }
+                    // package-level override
+                    (ns, packages) => {
+                        for pkg in packages {
+                            wkg_config_overrides.insert(
+                                format!("{ns}:{pkg}"),
+                                Override {
+                                    path: Some(path.clone()),
+                                    version: None,
+                                },
+                            );
+                        }
+                    }
+                }
+            }
+            RegistryPullSource::Builtin => bail!("no builtins are supported"),
+            RegistryPullSource::RemoteHttp(s) => {
+                let url = Url::parse(s)
+                    .with_context(|| format!("invalid registry pull source url [{s}]"))?;
+                let tempdir = tempfile::tempdir()
+                    .with_context(|| format!("failed to create temp dir for downloading [{url}]"))?
+                    .into_path();
+                let output_path = tempdir.join("unpacked");
+                let http_client = wash_lib::start::get_download_client()?;
+                let req = http_client.get(url.clone()).send().await.with_context(|| {
+                    format!("failed to retrieve WIT output from URL [{}]", &url)
+                })?;
+                let mut archive = tokio_tar::Archive::new(GzipDecoder::new(
```

Before this step, we should probably check the content-type header to check if the incoming type is `application/x-gzip` or `application/gzip` so we can fail early with a friendly message.

---

I don't think it's a given that this will be properly set -- what I *did* consider was checking the magic bytes of the downloaded bytes, BUT I think that's sufficiently covered by the error message when we try to unpack it. 

If things fail to unpack, the user *will* get a nice error message, including the URL we tried to download it from.